api/src/main/java/org/pmiops/workbench/ras/OpenIdConnectClient.java

```diff
@@ -0,0 +1,96 @@
+package org.pmiops.workbench.ras;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.auth.oauth2.AuthorizationCodeFlow;
+import com.google.api.client.auth.oauth2.BearerToken;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.auth.oauth2.StoredCredential;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.http.BasicAuthentication;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.MemoryDataStoreFactory;
+import java.io.IOException;
+import java.util.Set;
+
+/**
+ * Wrapper class to make <a href="https://openid.net/specs/openid-connect-core-1_0.html">OpenId
+ * Connect call</a> using Google OAuth Library.
+ */
+public class OpenIdConnectClient {
+  private static final NetHttpTransport DEFAULT_HTTP_TRANSPORT = new NetHttpTransport();
+  private static final JacksonFactory DEFAULT_JACKSON_FACTORY = new JacksonFactory();
+
+  private final AuthorizationCodeFlow codeFlow;
+  private final String userInfoUrl;
+  private final ObjectMapper objectMapper;
+
+  OpenIdConnectClient(
+      String clientId,
```

Comment:
This constructor signature seems very error-prone as a caller (5 strings in the exact correct order), can it be reduced at all? Aren't the token, authorize, and user info paths well known and could be constants in this class?

Reply:
Yep, this is nice - thank you.